crates/primitives/Cargo.toml

```diff
@@ -19,6 +19,7 @@ ink_prelude = { version = "3.1.0", path = "../prelude/", default-features = fals
 scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive", "full"] }
 scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
 cfg-if = "1"
+sha2-const = { version = "0.1.2", default-features = false }
```

Comment:
Why does it need to be a const implementation? You hopefully never call it at runtime but only from a proc macro (or in tests). So it can just be a normal blake2 runtime implementation here.

Reply:
The main place where we need that:
<img width="767" alt="image" src="https://user-images.githubusercontent.com/18346821/169512303-43596041-216b-4ca6-9d15-40da20820cfb.png">

We need to decide how the parent's key should affect the child's key:
- We can hash it, but that means we need to do that during compilation time. I selected that way.
- We can xor it, but it doesn't work in the: Parent -> Child -> Parent. xOr = Parent ^ Child ^ Parent = Child.
- We can use overflow and simply `+` storage key. But I'm not sure is that deterministic.

Reply:
Ahh okay so it is actually used in runtime code. I see.